packages/superset-ui-legacy-plugin-chart-sunburst/src/Sunburst.js

```diff
@@ -312,8 +311,10 @@ function Sunburst(element, props) {
         if (!isLeafNode) {
           // Not yet at the end of the sequence; move down the tree.
           let foundChild = false;
-          for (let k = 0; k < children.length; k++) {
-            currChild = children[k];
+
+          // eslint-disable-next-line unicorn/no-for-loop
+          for (let i = 0; i < children.length; i++) {
```

Comment:
If you did want to get rid of this for loop, you could probably also get rid of the `foundChild` variable with something like this:
```
childNode = children.filter(child => child.name === nodeName && child.level === level)[0]; // returns the first value or falsey undefined

if(!childNode){
  childNode = {
    name: nodeName,
    children: [],
    level,
  }
  children.push(childNode);
}
```

Reply:
Great idea. I made small change replacing `.filter` with `.find`